test/framework/cluster.go

```diff
@@ -94,6 +98,26 @@ type Provider interface {
 	Setup()
 }
 
+func (e *ClusterE2ETest) WithEKsaVersion(version *semver.Version) *ClusterE2ETest {
+	e.eksaBinaryLocation = e.GetReleaseFromVersion(version)
+	return e
+}
+
+func (e *ClusterE2ETest) WithLatestEksaVersion() *ClusterE2ETest {
+	e.eksaBinaryLocation = defaultEksaBinaryLocation
+	return e
+}
+
+func (e *ClusterE2ETest) WithLatestMinorReleaseFromMain() *ClusterE2ETest {
+	e.eksaBinaryLocation = e.GetLatestMinorReleaseBinaryFromMain()
+	return e
+}
+
+func (e *ClusterE2ETest) WithLatestMinorReleaseFromVersion(version *semver.Version) *ClusterE2ETest {
+	e.eksaBinaryLocation = e.GetLatestMinorReleaseBinaryFromReleaseBranch(version)
```

Comment:
I wonder if we should change this function name to `GetLatestMinorReleaseBinaryFromVersion` to be consistent with naming

Reply:
haha yeah we should....and I did this right after I suggested something similar to you

Reply:
hm yeah we could do that, since we are in fact looking at the semver as the primary indicator and not a branch name per se